### ISSUE TITLE
Doesn't work with vulkan and borderless/fullscreen, also only works with 1440p

### ISSUE THREAD

**Issue by xBeo** [seyahdoo/baldurs-gate-3-auto-join-conversation#2]
EDIT: See my [latest message](https://github.com/seyahdoo/baldurs-gate-3-auto-join-conversation/issues/2#issuecomment-1693355302) in this thread, there's more information.

Hi,

I'm so happy this tool exists, sadly it doesn't work as expected for me and my friends. I'm not sure why though, since everyone else seems to be happy with it.

A few days ago I downloaded the EXE for testing, startet it and put it on the second monitor. As soon as I opened the listen-in.png from the repo on my main monitor it worked. The day after I let my friends downlad it, and we tested it ingame. Sadly it didn't react. 
Just fyi, we have 1080p, 1440p and 4k screens respectively, windowed fullscreen and the EXE is on the second monitor. Sadly the programm didn't recognize any listen-in buttons ingame.

So, to try some things I clicked on the EXE on the second monitor and suddenly it recognized the button, clicked on it and I joined the dialouge. However, for some reason it continued "seeing" the button, even though, I already joined and the button wasn't there anymore. Clicking the window of the EXE again helped (sometimes), and it stopped. 
I tried recreating this, and it worked sporadically. Sometimes it did't react at all, sometimes it actually started and stopped clicking. For one of my friends (4k screen, sitting next to me) the behaviour was similar and inconsistent as well. For the other one (1080p) it didn't work at all, no idea if he was doing things wrong though.

Do you have any idea what could be the reason for that? Why would the captured screen (for the image regognition) be dependent on the focus of the EXE, and how does it "recognize" the button long after it disappeared? Why does it work with the example pictures?
Is there anyone who has similar behaviour?

I feel really dumb tbh, since everyone else seems to not have these issues.

EDIT: I just realized, there's a documentation now, and there's one step we definitly didn't do. Opening the pictures. Maybe that's the problem, even though I have no idea why that would be the case. Anyway, I'll update, as soon as I can test it (later this day).

**Comment by timcalire** [comment]
My .exe does not react at all to the game. When I compared both example pictures in source code to the ingame listen-in picture I discovered that resolution is different; ingame is smaller. 
Maybe that is why it does not work. Me and all my friends are playing at 1080p.

**Comment by xBeo** [comment]
Well, that at least matches my friends experience, who's playing 1080p as well. Also, my other friend just told me, that it actually didn't work at all for him (4k) either. Basically, from our experience, it only worked for me (1440p), and only with the aforementioned problems. I'll test it with the instructions from the other issue now and try a few things, we'll see if it changes anything.

**Comment by xBeo** [comment]
Sooooo, I found out quite a few things:

1) Opening the two images from the source code in the background (as described in the instructions) doesn't change anything at any point as far as i can tell. Not sure why that's a thing.

2) My problem was Vulkan (and borderless window mode). I chose Vulkan as my render engine and it looks like Vulkan does something with the image relayed to windows. When the listen-in button appears, nothing happens. When I cange focus from the game to windows, the button is recognized - again and again, even though it's not there anymore. As soon as I change focus again, it stops recognizing. _So somehow only the image in the moment of switching focus is relayed to the program._ I have no idea why, but that's clearly a windows/vulkan/nvidia?/whatever thing. The only way it works, is by choosing windowed (NOT borderless) mode in the game, which surprisingly works fine, it still doesn't have borders that way.
**So, the workaround is: Don't use Vulkan, or alternatively use windowed mode.**

3) Resolution is indeed a problem. 1440p works like a charm. Everything else seemingly doesn't. If I set the game to 1440p on my friends screen it works. His native 4k doesn't. Btw, the image from the source code is recognized, but as @timcalire already noted, it looks different than ingame. On my friends 4k screen it's small af, way smaller than ingame.

Is there a way to include other resolutions as well? As far as I can tell, we just need different resolution reference pictures, and some way to decide which one to use. I'd like to try it myself, but I have no idea how to compile and stuff, so yeah... I might give it a try at some point but that might take a while.

For the Vulkan stuff... No idea what's going on here, I guess I'll either use windowed or not use Vulkan at all. Who cares anyway.

**Comment by seyahdoo** [comment]
So I play the game on 1440p and the image of the button I got is from a 1440p screenshot, I will change the code so the button image is user editable. So you can get a screenshot from your game and use that instead.

**Comment by seyahdoo** [comment]
Changed the png to be user editable and did a release. If you would like to help, you may make the listen-in.png for the different resolutions of the game. So I can write some code to accomodate that. Before that happens this script will only support 1440p.

https://github.com/seyahdoo/baldurs-gate-3-auto-join-conversation/releases/tag/v1.0.4

**Comment by timcalire** [comment]
![listen-in](https://github.com/seyahdoo/baldurs-gate-3-auto-join-conversation/assets/143181039/23f7bbbd-0791-4b0b-8b6c-44813afefa7e)
Made listen-in for 1080p, working for me.

**Comment by xBeo** [comment]
Thank you very much!
Here's the listen-in PNGs for 1080p and 4k: https://seafile-public.mitkreutersalz.xyz/d/8b6cbad870174b558676/

We tested them yesterday, and it kinda worked too good. The button was always recognized, sometimes even in the texture of the ground or something like that. Decided a dialouge for us at one point ^^
Maybe the confidence is a bit too low (or high? no idea how it really works) now? 

Anyways, it's coming together, thank you for the great tool!

**Comment by seyahdoo** [comment]
Amazing, I'm glad its useful for you guys. 

Yes there is a confidence value on the code that governs how perfect the match should be to trigger the click. Currently Its hard set to 70 percent. I'll change the code again to have this user editable, so you guys can experiment with it.

**Comment by seyahdoo** [comment]
here it is:

https://github.com/seyahdoo/baldurs-gate-3-auto-join-conversation/releases/tag/v1.1.1

**Comment by NullChain0** [comment]
**Here's a few things that might be a takeaway** @xBeo 

--EDIT: On big screens with a 4K res. it might be best to UP the DPI on the desktop to increase GUI or inrceasing GUI size ingame. Vulkan indeed works different in capture processing so it might not work as its only been tested in directX so far. (maybe different with 1.1.1 - I run 1.0.3 will update later. Under this post are a few things explaining.) 

- Fullscreen might intervene with it. (Doubt this is true but it might decrease confidence slightly)
- vulkan captures images in its own independant way (This can seriously interfere as you have seen)
- FSR or anything else that might interact and make the image more stretched or slightly different.  (Anything that redraws images like NVIDIA DLSS or AMD FSR could decrease confidence the more it needs to work to get your target FPS)

All of these options might decrease the confidence of the EXE trying to recognize the images given.

FILLER INFO FOR DEV: @seyahdoo
Odd! This is probably solved already but I have a 4k resolution screen myself. 27 inch, might be that on bigger screens the image might stretch? I tested it in various DPIs from windows as well keep in mind this was in the previous version (version 1.0.3)

Here's my display information - Tested at 4k @ 27inch and that works as well! (both on 100-125-150%-175% & custom DPI) I didn't test higher as 200% looks a bit too big for that display so almost no one would use it. If you want I can continue testing though. (keep in mind ill do any further testing with 1.1.1)


(Mainscreen desktop1 NVIDIA GPU running 4K)
![rundll32_Cok6NzXGVU](https://github.com/seyahdoo/baldurs-gate-3-auto-join-conversation/assets/143140228/b813d729-0a75-4855-a4c0-91ca7ccdfa58)

![ApplicationFrameHost_LyFruaMSVj](https://github.com/seyahdoo/baldurs-gate-3-auto-join-conversation/assets/143140228/15b59e96-3390-49e4-a357-bc8d6f137ae8)

![ApplicationFrameHost_KVMmHZSR4V](https://github.com/seyahdoo/baldurs-gate-3-auto-join-conversation/assets/143140228/3e3c8d73-27fb-4367-8f12-7d55d2519e58)


I can upload the screen properties of desktop 2 (AMD GPU) if need be this will be 1440P tested at 100-125-150%

(Specifications of both desktops)
One runs NVIDIA - RTX 4080 - 13700K - DDR5 32GB
One runs AMD 5700XT - Ryzen 3600x - DDR4 64GB
(Included these just to add that both AMD & NVIDIA systems do work on my end (still version 1.0.3 at this time, will update later if any issues ill report back)

Related info: Both run DirectX as capture. NOT Vulkan nor have I tried this.


Let me know if anything else is needed. Figured I'd drop this as it might be relevant on seeing where the error might be if it wont be resolved with this update.
